pkg/rke2/registries_types.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2022 SUSE.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rke2
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	bootstrapv1 "github.com/rancher-sandbox/cluster-api-provider-rke2/bootstrap/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Mirror contains the config related to the registry mirror
+type Mirror struct {
+	// Endpoints are endpoints for a namespace. CRI plugin will try the endpoints
+	// one by one until a working one is found. The endpoint must be a valid url
+	// with host specified.
+	// The scheme, host and path from the endpoint URL will be used.
+	Endpoints []string `toml:"endpoint" yaml:"endpoint"`
```

Comment:
Struct field name is plural but the tag name is singular. Is that right?

Reply:
I agree, its best not to tie ourselves unless we have to. Personally, i find it weird that when marshalled its a singular name when the func field is plural. But thats just my preference.

Reply:
Since, we will be maintaining our own version of it, we could also change the field name and leave the json annotation, making both singular.

Reply:
I did just modify everywhere to singular.